cli/src/commands/pack.js

```diff
@@ -27,30 +27,32 @@ exports.builder = yargs =>
         })
 
 exports.handler = async argv => {
-    const { cwd = process.cwd(), folder, destination, filename } = argv
+    const { cwd = process.cwd(), source, destination, filename } = argv
 
     const paths = makePaths(cwd)
-    const config = fs.readJsonSync(paths.buildAppConfigJson)
+    const config = fs.readJsonSync(paths.buildAppConfigJson, { throws: false })
 
-    let inputPath, outputPath
-    if (config.type === 'app') {
-        inputPath = path.resolve(cwd, folder ? folder : paths.buildAppOutput)
+    if (!config) {
+        // we may be dealing with a library or a unbuilt app
+        // load the d2.config.js file from the project and check
+        const baseConfig = require(paths.config)
```

Comment:
I think this will throw if the config file doesn't exist (i.e. if used on build output from a non-platform app) which I think would be a really useful feature (packing a directory is currently a big pain because you need to make absolutely sure you make a flat zip etc.  I'll add a check for file existence here.